src/Compilers/Core/Portable/SourceFileResolver.cs

```diff
@@ -39,21 +48,65 @@ public SourceFileResolver(ImmutableArray<string> searchPaths, string baseDirecto
 
             _baseDirectory = baseDirectory;
             _searchPaths = searchPaths;
-        }
+            _pathMap = pathMap;
 
-        public string BaseDirectory
-        {
-            get { return _baseDirectory; }
+            // the keys in pathMap should not end with a path separator
+            if (!pathMap.IsDefaultOrEmpty)
+            {
+                foreach (var kv in pathMap)
+                {
+                    var key = kv.Key;
+                    if (key == null || key.Length == 0)
+                    {
+                        throw new ArgumentException(CodeAnalysisResources.EmptyKeyInPathMap, nameof(pathMap));
+                    }
+
+                    if (IsPathSeparator(key[key.Length - 1]))
```

Comment:
I don't think so. The worst that will happen is that you won't be happy with the shape of the file names appearing in the compiler output.